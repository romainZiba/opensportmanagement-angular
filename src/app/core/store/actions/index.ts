export * from './layout.actions';
export * from './teams.actions';
