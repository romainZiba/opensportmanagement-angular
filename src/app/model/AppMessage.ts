export class AppMessage {
  body: string;
  authorUserName: string;
  authorFirstName: string;
  authorLastName: string;
  conversationId: string;
  recipients: number[];
  conversationTopic: string;
  creationDate: string;
}
