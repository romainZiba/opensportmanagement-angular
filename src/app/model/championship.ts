export interface IChampionship {
  name: string;
}

export class Championship {
  name: string;
  _id: number;
}
