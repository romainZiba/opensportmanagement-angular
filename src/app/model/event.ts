import {TeamMember} from './team-member';

export class Event {
  _id: number;
  name: string;
  description: string;
  fromDateTime: string;
  toDateTime: string;
  placeId: number;
  placeName: string;
  done: boolean;
  localTeamName: string;
  localTeamImgUrl: string;
  localTeamScore: number;
  visitorTeamName: string;
  visitorTeamImgUrl: string;
  visitorTeamScore: number;
  presentMembers: TeamMember[];
  absentMembers: TeamMember[];
  waitingMembers: TeamMember[];
  cancelled: boolean;
  openForRegistration: boolean;
}

export class EventCreateUpdate {
  teamId: number;
  name: string;
  fromDate: string;
  toDate: string;
  fromTime: string;
  toTime: string;
  placeId: number;
  isRecurrent: boolean;
  recurrenceDays: string[];
  opponentId: number;
  championshipId: number;
  type: EventType;
}

export enum EventType {
  TRAINING = "TRAINING",
  MATCH = "MATCH",
  OTHER = "OTHER"
}
