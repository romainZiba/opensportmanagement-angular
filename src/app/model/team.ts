export class Team {
  _id: number;
  name: string;
  sport: string;
  genderKind: string;
  ageGroup: string;
  imgUrl: string;
}
