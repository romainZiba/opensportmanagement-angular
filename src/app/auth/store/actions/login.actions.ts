import {Action} from '@ngrx/store';
import {Credentials, User} from '../../models/user';

export enum LoginActionsType {
  LOGIN = '[Auth] Login',
  LOGIN_SUCCESS = '[Auth] Login Success',
  LOGIN_FAILURE = '[Auth] Login Failure',
  LOGIN_REDIRECT = '[Auth] Login Redirect',
  LOGOUT = '[Auth] Logout',
  CHECK_TOKEN_VALID = '[Auth] Check token valid'
}

export class Login implements Action {
  readonly type = LoginActionsType.LOGIN;
  constructor(public payload: Credentials) {}
}

export class LoginSuccess implements Action {
  readonly type = LoginActionsType.LOGIN_SUCCESS;
  constructor(public payload: User) {}
}

export class LoginFailure implements Action {
  readonly type = LoginActionsType.LOGIN_FAILURE;
  constructor(public payload: any) {}
}

export class LoginRedirect implements Action {
  readonly type = LoginActionsType.LOGIN_REDIRECT;
}

export class Logout implements Action {
  readonly type = LoginActionsType.LOGOUT;
}

export class CheckTokenValid implements Action {
  readonly type = LoginActionsType.CHECK_TOKEN_VALID
}

export type LoginActions =
  | Login
  | LoginSuccess
  | LoginFailure
  | LoginRedirect
  | Logout
  | CheckTokenValid;
