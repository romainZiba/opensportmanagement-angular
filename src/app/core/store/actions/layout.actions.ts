import { Action } from '@ngrx/store';

export enum SidenavActionsType {
  OPEN_SIDENAV = '[Core] Open Sidenav',
  CLOSE_SIDENAV = '[Core] Close Sidenav',
}

export class OpenSidenav implements Action {
  readonly type = SidenavActionsType.OPEN_SIDENAV;
}

export class CloseSidenav implements Action {
  readonly type = SidenavActionsType.CLOSE_SIDENAV;
}

export type SidenavActions = OpenSidenav | CloseSidenav;
