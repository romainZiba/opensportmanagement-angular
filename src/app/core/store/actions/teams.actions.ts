import { Action } from "@ngrx/store";
import { List } from "immutable";
import { Team } from "../../../model/team";

export enum TeamsActionsType {
  LOAD_TEAMS = "[Core] Load teams",
  LOAD_TEAMS_SUCCESS = "[Core] Load teams success",
  LOAD_TEAMS_FAILED = "[Core] Load teams failed",
  SELECT_TEAM = "[Core] Select team"
}

export class LoadTeams implements Action {
  readonly type = TeamsActionsType.LOAD_TEAMS;
}

export class LoadTeamsSuccess implements Action {
  readonly type = TeamsActionsType.LOAD_TEAMS_SUCCESS;
  constructor(public payload: List<Team>) {}
}

export class LoadTeamsFailed implements Action {
  readonly type = TeamsActionsType.LOAD_TEAMS_FAILED;
  constructor(public payload: any) {}
}

export class SelectTeam implements Action {
  readonly type = TeamsActionsType.SELECT_TEAM;
  constructor(public payload: number) {}
}

export type TeamsActions =
  | LoadTeams
  | LoadTeamsSuccess
  | LoadTeamsFailed
  | SelectTeam;
