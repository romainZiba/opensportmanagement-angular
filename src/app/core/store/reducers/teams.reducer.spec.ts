import { initialState, reducer } from "./teams.reducer";
import * as teamsActions from "../actions";
import { List } from "immutable";
import { Team } from "../../../model/team";

describe("TeamsReducer", () => {
  const team1: Team = {
    _id: 1,
    name: "team #1",
    sport: "basketball",
    genderKind: "MALE",
    ageGroup: "ADULTS",
    imgUrl: ""
  };

  const team2: Team = {
    _id: 2,
    name: "team #2",
    sport: "football",
    genderKind: "FEMALE",
    ageGroup: "ADULTS",
    imgUrl: ""
  };

  const team3: Team = {
    _id: 3,
    name: "team #3",
    sport: "football",
    genderKind: "MALE",
    ageGroup: "ADULTS",
    imgUrl: ""
  };

  const teams: List<Team> = List([team1, team2, team3]);

  const entities = {
    1: teams.get(0),
    2: teams.get(1),
    3: teams.get(2)
  };

  describe("undefined action", () => {
    it("should return the default state", () => {
      const action = {} as any;
      const result = reducer(undefined, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe("Load teams", () => {
    it("should return loading", () => {
      const action = new teamsActions.LoadTeams();
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe("Load teams success", () => {
    it("should return loaded with entities", () => {
      const action = new teamsActions.LoadTeamsSuccess(teams);
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe("Load teams failed", () => {
    it("should return loaded false", () => {
      const error = "error";
      const action = new teamsActions.LoadTeamsFailed(error);
      const result = reducer({ ...initialState, loading: true }, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe("Select team", () => {
    it("should return selected team", () => {
      const action = new teamsActions.SelectTeam(1);
      const result = reducer(
        { ...initialState, entities, loaded: true },
        action
      );
      expect(result).toMatchSnapshot();
    });
  });
});
