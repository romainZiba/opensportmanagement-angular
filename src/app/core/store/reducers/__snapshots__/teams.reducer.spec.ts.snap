// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamsReducer Load teams failed should return loaded false 1`] = `
Object {
  "entities": Object {},
  "loaded": false,
  "loading": false,
  "selectedTeam": null,
}
`;

exports[`TeamsReducer Load teams should return loading 1`] = `
Object {
  "entities": Object {},
  "loaded": false,
  "loading": true,
  "selectedTeam": null,
}
`;

exports[`TeamsReducer Load teams success should return loaded with entities 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "_id": 1,
      "ageGroup": "ADULTS",
      "genderKind": "MALE",
      "imgUrl": "",
      "name": "team #1",
      "sport": "basketball",
    },
    "2": Object {
      "_id": 2,
      "ageGroup": "ADULTS",
      "genderKind": "FEMALE",
      "imgUrl": "",
      "name": "team #2",
      "sport": "football",
    },
    "3": Object {
      "_id": 3,
      "ageGroup": "ADULTS",
      "genderKind": "MALE",
      "imgUrl": "",
      "name": "team #3",
      "sport": "football",
    },
  },
  "loaded": true,
  "loading": false,
  "selectedTeam": null,
}
`;

exports[`TeamsReducer Select team should return selected team 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "_id": 1,
      "ageGroup": "ADULTS",
      "genderKind": "MALE",
      "imgUrl": "",
      "name": "team #1",
      "sport": "basketball",
    },
    "2": Object {
      "_id": 2,
      "ageGroup": "ADULTS",
      "genderKind": "FEMALE",
      "imgUrl": "",
      "name": "team #2",
      "sport": "football",
    },
    "3": Object {
      "_id": 3,
      "ageGroup": "ADULTS",
      "genderKind": "MALE",
      "imgUrl": "",
      "name": "team #3",
      "sport": "football",
    },
  },
  "loaded": true,
  "loading": false,
  "selectedTeam": 1,
}
`;

exports[`TeamsReducer undefined action should return the default state 1`] = `
Object {
  "entities": Object {},
  "loaded": false,
  "loading": false,
  "selectedTeam": null,
}
`;
